src/pages/dashboard/recipe-run/[id].js

```diff
@@ -46,12 +60,15 @@ const RecipeRun = () => {
       </Layout>
     )
   }
-  if (!recipeRun)
+  if (!recipeRun || !meta || !prefect || !spec) {
```

Comment:
If we expect the feedstock metadata to be already fetched much of the time (when navigating from the feedstock page), I wonder if showing the feedstock information with a loading state specifically in the recipe run section would feel nicer.

That would fork this loading logic into something like:
```
1. if !meta || !prefect || !spec -> show current loading state
2. else if !recipeRun -> render feedstock info, show recipe run loading state
```

Reply:
i like this idea. I'm going to tinker with this and will ping you if i need help :)

Reply:
the latest iteration addresses this. See https://pangeo-forge-3cv0gj86i-pangeo-forge.vercel.app/